core/domain/translation_domain.py

```diff
@@ -18,9 +18,262 @@
 
 from __future__ import annotations
 
+import enum
+
+from core import feconf
 from core import utils
 
-from typing import Dict
+from typing import Dict, List
+from typing_extensions import TypedDict
+
+
+class TranslatableContentFormat(enum.Enum):
+    """Represents all possible data types for any translatable content."""
+
+    HTML = 'html'
+    UNICODE_STRING = 'unicode'
+    SET_OF_NORMALIZED_STRING = 'set_of_normalized_string'
+    SET_OF_UNICODE_STRING = 'set_of_unicode_string'
+
+
+class TranslatableContentDict(TypedDict):
+    """Dictionary representing TranslatableContent object."""
+
+    content_id: str
+    content: feconf.ContentInTranslatableContent
+    content_type: TranslatableContentFormat
+
+
+class TranslatableContentsCollection:
+    """A class to collect all TranslatableContents from a translatable object
+    and maps with their corresponding content-ids.
+    """
+
+    def __init__(self) -> None:
+        """Constructs a TranslatableContentsCollection object."""
+        self.translatable_contents: Dict[str, TranslatableContent] = {}
+
+    def add_translatable_field(
+        self,
+        field_type: TranslatableContentFormat,
+        content_id: str,
+        value: feconf.ContentInTranslatableContent
+    ) -> None:
+        """Adds translatable field parameter to translatable_contents dict.
+
+        Args:
+            field_type: TranslatableContentFormat. The type of the
+                corresponding translatable content.
+            content_id: str. The id of the corresponding translatable content.
+            value: ContentInTranslatableContent. Value of the content which is
+                translatable.
+        """
+        self.translatable_contents[content_id] = TranslatableContent(
+            content_id, value, field_type)
+
+    def add_translatable_object(
+        self,
+        value: BaseTranslatableObject
```

Comment:
Perhaps change value --> translatable_object

EDIT: though, see above; since you only need the translatable_contents, probably better for TranslatableContentsCollection to operate only with other TranslatableContentsCollections.